File: backend/src/main/java/com/backend/domain/user/repository/RefreshRedisRepository.java
package com.backend.domain.user.repository;


import com.backend.domain.user.entity.RefreshToken;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Repository;

import java.util.concurrent.TimeUnit;

@Repository
@RequiredArgsConstructor
public class RefreshRedisRepository {

    private final RedisTemplate<String, Object> redisTemplate;

    private static final String REFRESH_TOKEN_PREFIX = "refreshToken_";

    public void save(RefreshToken refreshToken) {
        redisTemplate.opsForValue().set(
                REFRESH_TOKEN_PREFIX + refreshToken.getUserId(),
                refreshToken.getRefreshToken(),
                refreshToken.getExpiration(),
                TimeUnit.SECONDS
                );
    }

    public String findByUserId(Long userId) {
        return (String) redisTemplate.opsForValue().get(REFRESH_TOKEN_PREFIX + userId);
    }

    public void deleteByUserId(Long userId){
        redisTemplate.delete(REFRESH_TOKEN_PREFIX + userId);
    }

    public boolean existsByUserId(Long userId) {
        return redisTemplate.hasKey(REFRESH_TOKEN_PREFIX + userId);
    }
}

File: backend/src/main/java/com/backend/domain/user/repository/RefreshRedisRepository.kt
package com.backend.domain.user.repository

import com.backend.domain.user.entity.RefreshToken
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Repository
import java.util.concurrent.TimeUnit


@Repository
class RefreshRedisRepository(
    private val redisTemplate: RedisTemplate<String, Any>
) {

    companion object {
        private const val REFRESH_TOKEN_PREFIX = "refreshToken_"
    }

    fun save(refreshToken: RefreshToken) {
        redisTemplate.opsForValue().set(
            REFRESH_TOKEN_PREFIX + refreshToken.userId,
            refreshToken.refreshToken,
            refreshToken.expiration,
            TimeUnit.SECONDS
        )
    }

    fun findByUserId(userId: Long): String? =
        redisTemplate.opsForValue().get(REFRESH_TOKEN_PREFIX + userId) as String?


    fun deleteByUserId(userId: Long) =
        redisTemplate.delete(REFRESH_TOKEN_PREFIX + userId)


    fun existsByUserId(userId: Long?): Boolean =
        redisTemplate.hasKey(REFRESH_TOKEN_PREFIX + userId)

}
